JMMServer/Repositories/BaseCachedRepository.cs

```diff
@@ -115,7 +126,7 @@ public virtual void Delete(T cr)
                 EndDeleteCallback?.Invoke(cr);
             }
         }
-        public virtual void Delete(List<T> objs)
+        public virtual void Delete(IReadOnlyCollection<T> objs)
```

Comment:
First of all, great work, So far, so good i'm only worried about the use of IReadOnlyCollection on the Save and Delete function of base Repository, the save collection function is a helper in fact, it should accept IEnumerable instead List, I only used list because of the free count. This force the user to convert any Enumerable into IReadOnlyCollection and i don't see any performance gain of doing so.